File: chazm-model/src/main/java/runtimemodels/chazm/model/event/HasEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.relation.Has;

/**
 * The {@linkplain HasEventFactory} interface defines the API for constructing {@linkplain HasEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface HasEventFactory {

    /**
     * Constructs an {@linkplain HasEvent}.
     *
     * @param category the {@linkplain EventType}.
     * @param has      the {@linkplain Has}.
     * @return a {@linkplain HasEvent}.
     */
    HasEvent build(EventType category, Has has);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/HasEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.relation.Has

/**
 * The [HasEventFactory] interface defines the API for constructing [HasEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface HasEventFactory {

    /**
     * Constructs an [HasEvent].
     *
     * @param category the [EventType].
     * @param has      the [Has].
     * @return a [HasEvent].
     */
    fun build(category: EventType, has: Has): HasEvent

}
